src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1731,6 +1731,14 @@ public function preferSafeContent(): bool
         return $this->isSafeContentPreferred = AcceptHeader::fromString($this->headers->get('Prefer'))->has('safe');
     }
 
+    /**
+     * Reset the mappings of formats to mime types.
+     */
+    public static function resetFormats(): void
+    {
+        static::$formats = null;
```

Comment:
```suggestion
        static::initializeFormats();
```

This is a micro-optimization, but this will save some time when handling the next request.

Reply:
@dunglas I considered doing it this way originally but my feeling was because the initial state of `$formats` is `null`, that's what we should set it back to as a reset., because that's the change that interferes least with the implementation details of the rest of the class. Happy to change it in accordance with your suggestion if you like.